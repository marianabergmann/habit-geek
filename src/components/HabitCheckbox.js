import React, { Fragment } from "react";
import "./HabitCheckbox.scss";

const HabitCheckbox = () => {
  return (
    <Fragment>
      <label class="toggle">
        <input class="toggle__input" type="checkbox" />
        <span class="toggle__label">
          <span class="toggle__text"></span>
        </span>
      </label>
    </Fragment>
  );
};

export default HabitCheckbox;
