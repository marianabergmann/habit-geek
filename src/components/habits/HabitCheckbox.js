import React, { Fragment } from "react";

import "./HabitCheckbox.scss";

const HabitCheckbox = ({ isChecked, clickHandler, isDisabled }) => {
  return (
    <Fragment>
      <label className="toggle">
        <input
          className="toggle__input"
          onChange={clickHandler}
          type="checkbox"
          checked={isChecked}
          disabled={isDisabled}
        />
        <span className="toggle__label">
          <span className="toggle__text"></span>
        </span>
      </label>
    </Fragment>
  );
};

export default HabitCheckbox;
