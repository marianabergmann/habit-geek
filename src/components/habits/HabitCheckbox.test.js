import React from "react";
import { shallow } from "enzyme";
import HabitCheckbox from "./HabitCheckbox";

describe("HabitCheckbox component", () => {
  it("Renders static elements", () => {
    const habitCheckbox = shallow(<HabitCheckbox />);
    expect(habitCheckbox).toMatchSnapshot();
  });

  it("renders checked checkboxes", () => {
    const habitCheckbox = shallow(<HabitCheckbox isChecked={true} />);
    expect(habitCheckbox).toMatchSnapshot();
  });

  it("Toggles the checkbox", () => {
    const clickHandlerMock = jest.fn();
    const dashboard = shallow(
      <HabitCheckbox clickHandler={clickHandlerMock} />
    );
    dashboard.find("input").simulate("click");
    expect(clickHandlerMock).toHaveBeenCalledTimes(1);
  });
});
