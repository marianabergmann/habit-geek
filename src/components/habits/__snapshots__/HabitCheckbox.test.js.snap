// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitCheckbox component Renders static elements 1`] = `
<Fragment>
  <label
    className="toggle"
  >
    <input
      className="toggle__input"
      type="checkbox"
    />
    <span
      className="toggle__label"
    >
      <span
        className="toggle__text"
      />
    </span>
  </label>
</Fragment>
`;

exports[`HabitCheckbox component renders checked checkboxes 1`] = `
<Fragment>
  <label
    className="toggle"
  >
    <input
      checked={true}
      className="toggle__input"
      type="checkbox"
    />
    <span
      className="toggle__label"
    >
      <span
        className="toggle__text"
      />
    </span>
  </label>
</Fragment>
`;
